metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key. Note defaults to string if type is not found. This is because most
+// types are strings and there are a lot of different metafield types that would need to be caught.
+func (m Metafields) GetByKey(key string) (interface{}, error) {
```

Comment:
Is there no alternative way that means we do not need to use `interface`? :thinking:

Reply:
Not before 1.18 unfortunately. I think when we bump the Shopify API version up (which will be inevitable). We will upgrade the GO version also and then can change the way the package works maybe.

Maintaining this package is of utmost importance because the whole business relies on our Shopify packages.